### PR TITLE
Disable docker save with image IDs

### PR DESCRIPTION
**Pull request by shishir-a412ed**
Signed-off-by: Shishir Mahajan shishir.mahajan@redhat.com

### CODE REVIEW

**Comment by shishir-a412ed** [comment]
Fixes #10592 

@tiborvass if you are okay with the error message. 
I ll add more details in the man page.

Let me know if there are other places I need to add documentation for this.

Shishir

**Comment by tiborvass** [comment]
@shishir-a412ed the "Fixes #XXX" has to be in the PR description (first comment in the thread), not in a comment after.

File: graph/export.go [hunk 1]
```diff
@@ -73,10 +74,8 @@ func (s *TagStore) ImageExport(imageExportConfig *ImageExportConfig) error {
 				}
 
 			} else {
-				// this must be an ID that didn't get looked up just right?
-				if err := s.exportImage(name, tempdir); err != nil {
-					return err
-				}
+				// this must be an ID
+				return fmt.Errorf("docker save by image id (%s) is not allowed. See man page for more details.", name)
```
Comment (tiborvass) [inline graph/export.go]: replace error message with `Cannot save image from image ID. Please use image name instead.`

File: graph/tags.go [hunk 2]
```diff
@@ -138,17 +160,6 @@ func (store *TagStore) LookupImage(name string) (*image.Image, error) {
...
-	}
-
-	// name must be an image ID.
-	store.Lock()
-	defer store.Unlock()
-	if img, err = store.graph.Get(name); err != nil {
-		return nil, err
-	}
-
 	return img, nil
```
Comment (tiborvass) [inline graph/tags.go]: `return store.GetImage(repoName, ref)`

**Comment by crosbymichael** [comment]
This breaks existing functionality.  This is an issue with load and overwriting existing data, not with saving.

**Pull request closed without merging**